karyon3-core/src/main/java/com/netflix/karyon/governator/AutoBindModule.java

```diff
@@ -0,0 +1,197 @@
+package com.netflix.karyon.governator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.Lists;
+import com.google.common.reflect.ClassPath;
+import com.google.inject.AbstractModule;
+import com.google.inject.Module;
+import com.google.inject.util.Modules;
+import com.netflix.governator.annotations.AutoBindSingleton;
```

Comment:
Good point.  I'm reluctant however to add too many sub-projects here.  At some point we'll just go through a regular deprecation cycle for AutoBindSingleton after which we will either remove this class or move it to a sub-project at that point.